src/pkg/repository/repository_test.go

```diff
@@ -236,6 +236,37 @@ func (suite *RepositoryIntegrationSuite) TestConnect() {
 	assert.NoError(t, err, clues.ToCore(err))
 }
 
+func (suite *RepositoryIntegrationSuite) TestRepository_UpdatePassword() {
+	t := suite.T()
+
+	ctx, flush := tester.NewContext(t)
+	defer flush()
+
+	// need to initialize the repository before we can test connecting to it.
+	st := storeTD.NewPrefixedS3Storage(t)
+	r, err := New(
+		ctx,
+		account.Account{},
+		st,
+		control.DefaultOptions(),
+		NewRepoID)
+	require.NoError(t, err, clues.ToCore(err))
+
+	err = r.Initialize(ctx, ctrlRepo.Retention{})
+	require.NoError(t, err, clues.ToCore(err))
+
+	// now re-connect
+	err = r.Connect(ctx)
+	assert.NoError(t, err, clues.ToCore(err))
+
+	err = r.UpdatePassword(ctx, "newpass")
+	require.NoError(t, err, clues.ToCore(err))
+
+	// now reconnect with new pass
+	err = r.Connect(ctx)
```

Comment:
I'm not sure this is doing what we hope it is. I think it would be safer to use a separate repo instance, especially since the repo struct [caches](https://github.com/alcionai/corso/blob/2fe0e8643e39c59a880d68ad4c0045b52dd23f39/src/pkg/repository/repository.go#L244) the storage config which itself [contains](https://github.com/alcionai/corso/blob/2fe0e8643e39c59a880d68ad4c0045b52dd23f39/src/pkg/storage/testdata/storage.go#L47) the [repo passphrase](https://github.com/alcionai/corso/blob/2fe0e8643e39c59a880d68ad4c0045b52dd23f39/src/pkg/storage/testdata/storage.go#L78)

I'm not really a fan of the below, but I believe it should allow updating an existing storage struct with a new password
```go
st.Config[storeTD.CorsoPassphrase] = "newpass"
```

Reply:
passphrase is saved in - `common_corsoPassphrase` and its not exposed mostly because of security reasons. And not sure if we should touch it. Also, the integration test in coming PRs will cover the other part of this test-case. As of know I will move with it. And if we are not satisfied with the testing, I will work on improving it.

Reply:
Given that this is just a test case it should be ok from a security perspective to mess with it. Not my favorite design paradigm, but at least it works

Reply:
Done